src/ng2-smart-table/components/cell/edit-cell.component.ts

```diff
@@ -10,9 +10,31 @@ export class EditCellComponent {
 
   @Input() cell: Cell;
   @Input() inputClass: string = '';
+  @ViewChild('dynamicTarget', {read: ViewContainerRef}) dynamicTarget: any;
+  customComponent: any;
 
   @Output() public edited: EventEmitter<any> = new EventEmitter<any>();
 
+  constructor(private resolver: ComponentFactoryResolver){
+  }
+
+  ngOnChanges(arg) {
```

Comment:
This also should be moved into the separate CustomEditorComponent so that we don't have this logic here. Moreover, we need to rename the cell types components into cell editors to follow the idea of cell editors:
1. https://github.com/akveo/ng2-smart-table/tree/master/src/ng2-smart-table/components/cell/cell-types cell-types -> cell-editors
2. cell-completer.component.ts -> completer-editor.component.ts
etc